tedium.js

```diff
@@ -181,12 +228,52 @@ function connectToGithub() {
   return github;
 }
 
+function analyzeRepos() {
+  const dirs = fs.readdirSync('repos/');
+  const htmlFiles = [];
 
+  for (const dir of dirs) {
+    for (const fn of fs.readdirSync(path.join('repos', dir))) {
+      if (/demo|index\.html|dependencies\.html/.test(fn) ||
+          !fn.endsWith('.html')) {
+        continue;
+      }
+      htmlFiles.push(path.join('repos', dir, fn));
+    }
+  }
 
-let user;
+  function filter(repo) {
+    return !cleanup.existsSync(repo);
+  }
 
-console.log('Discovering repos in PolymerElements...');
+  // This code is conceptually simple, it's only complex due to ordering
+  // and the progress bar. Basically we call analyzer.metadataTree on each
+  // html file in sequence, then finally call analyzer.annotate() and return.
+  return hydrolysis.Analyzer.analyze('repos/polymer/polymer.html', {filter})
+    .then((analyzer) => {
+      let promise = Promise.resolve();
+      const progressBar = new ProgressBar(`:msg [:bar] :percent`, {
+        total: htmlFiles.length + 1, width: progressBarWidth});
+
+      for (const htmlFile of htmlFiles) {
+        promise = promise.then(() => {
+          return analyzer.metadataTree(htmlFile).then(() => {
+            const msg = pad(`Analyzing ${htmlFile.slice(6)}`,
+                            progressMessageWidth, {strip: true});
+            progressBar.tick({msg});
+          });
+        });
+      }
+      return promise.then(() => {
+        progressBar.tick({
+            msg: pad('Analyzing with hydrolysis...', progressMessageWidth)});
+        analyzer.annotate();
+        return analyzer;
+      });
+    });
+}
 
+let user;
 
 Promise.resolve().then(() => {
```

Comment:
I initially had that! Then I discovered that promisify doesn't use standard promises, but instead returns Q promises or something (maybe bluebird?). Specifically, those promises didn't have one of the standard methods.. catch I think?

Wrapping with Promise.resolve().then() means that those promises get coerced into standard promises.